src/components/jobManager.ts

```diff
@@ -168,9 +168,15 @@ export class JobManager {
 					continue;
 				}
 
+				const deletedJobs = new Set(
```

Comment:
why is this check needed?

Reply:
i wanted to pass jobs instead of hashes so i can group jobs by executionId in the tracking service.